lib/linter/node-event-generator.js

```diff
@@ -9,248 +9,28 @@
-			return getPossibleTypes(parsedSelector.right);
-
-		case "class":
-			if (parsedSelector.name === "function") {
-				return [
-					"FunctionDeclaration",
-					"FunctionExpression",
-					"ArrowFunctionExpression",
-				];
-			}
-
-			return null;
-
-		default:
-			return null;
-	}
-}
-
-/**
- * Counts the number of class, pseudo-class, and attribute queries in this selector
- * @param {Object} parsedSelector An object (from esquery) describing the selector's matching behavior
- * @returns {number} The number of class, pseudo-class, and attribute queries in this selector
- */
-function countClassAttributes(parsedSelector) {
-	switch (parsedSelector.type) {
-		case "child":
-		case "descendant":
-		case "sibling":
-		case "adjacent":
-			return (
-				countClassAttributes(parsedSelector.left) +
-				countClassAttributes(parsedSelector.right)
-			);
-
-		case "compound":
-		case "not":
-		case "matches":
-			return parsedSelector.selectors.reduce(
-				(sum, childSelector) =>
-					sum + countClassAttributes(childSelector),
-				0,
-			);
-
-		case "attribute":
-		case "field":
-		case "nth-child":
-		case "nth-last-child":
-			return 1;
-
-		default:
-			return 0;
-	}
-}
-
-/**
- * Counts the number of identifier queries in this selector
- * @param {Object} parsedSelector An object (from esquery) describing the selector's matching behavior
- * @returns {number} The number of identifier queries
- */
-function countIdentifiers(parsedSelector) {
-	switch (parsedSelector.type) {
-		case "child":
-		case "descendant":
-		case "sibling":
-		case "adjacent":
-			return (
-				countIdentifiers(parsedSelector.left) +
-				countIdentifiers(parsedSelector.right)
-			);
-
-		case "compound":
-		case "not":
-		case "matches":
-			return parsedSelector.selectors.reduce(
-				(sum, childSelector) => sum + countIdentifiers(childSelector),
-				0,
-			);
-
-		case "identifier":
-			return 1;
-
-		default:
-			return 0;
-	}
-}
+//-----------------------------------------------------------------------------
 
 /**
- * Compares the specificity of two selector objects, with CSS-like rules.
- * @param {ASTSelector} selectorA An AST selector descriptor
- * @param {ASTSelector} selectorB Another AST selector descriptor
- * @returns {number}
- * a value less than 0 if selectorA is less specific than selectorB
- * a value greater than 0 if selectorA is more specific than selectorB
- * a value less than 0 if selectorA and selectorB have the same specificity, and selectorA <= selectorB alphabetically
- * a value greater than 0 if selectorA and selectorB have the same specificity, and selectorA > selectorB alphabetically
+ * Compares two ESQuery selectors by specificity.
+ * @param {ESQueryParsedSelector} a The first selector to compare.
+ * @param {ESQueryParsedSelector} b The second selector to compare.
+ * @returns {number} A negative number if `a` is more specific than `b`, a positive number if `b` is more specific than `a`, or zero if they are equally specific.
```

Comment:
```suggestion
 * @returns {number} A negative number if `a` is less specific than `b` or they are equally specific and `a` <= `b` alphabetically, a positive number if `a` is more specific than `b`.
```

This never returns zero, and returns a negative number if `a` is less (not more) specific than `b`.